tests/test_validate.py

```diff
@@ -768,3 +768,94 @@ def test_that_standardize_string_for_validation_returns_expected(
         allow_underscore=allow_underscore,
     )
     assert test_str == expected
+
+
+@pytest.mark.parametrize(
+    "input,expected_index,allowed_alleles,ignore_case",
+    [
+        (
+            pd.DataFrame(
+                {"REFERENCE_ALLELE": ["ACGT-G", "A-CGT ", "A", "C", "T", "G", "-", " "]}
+            ),
+            pd.Index([]),
+            ["A", "T", "C", "G", " ", "-"],
+            True,
+        ),
+        (
+            pd.DataFrame({"REFERENCE_ALLELE": ["acgt-g", "acgt", "  "]}),
+            pd.Index([]),
+            ["A", "T", "C", "G", " ", "-"],
+            True,
+        ),
+        (
+            pd.DataFrame({"REFERENCE_ALLELE": ["@##", "ACGTX"]}),
+            pd.Index([0, 1]),
+            ["A", "T", "C", "G", " ", "-"],
+            True,
+        ),
+        (
+            pd.DataFrame({"REFERENCE_ALLELE": ["XXX", "ACGT"]}),
+            pd.Index([0]),
+            ["A", "T", "C", "G", " ", "-"],
+            True,
+        ),
+        (
+            pd.DataFrame({"REFERENCE_ALLELE": ["ACGT-G", pd.NA, None]}),
```

Comment:
Are `float('nan')` currently allowed in the maf?

Reply:
According to [this code](https://github.com/Sage-Bionetworks/Genie/blob/develop/genie_registry/maf.py#L161-L171), it seems like it could be possible for allele columns to be read in as numeric and get something like `float('nan')` (i believe this is only for numeric columns). 

This makes me wonder if we should have standardization of columns and their datatypes before hand (either they have the wrong datatype and stop validation there, or they have the wrong datatype, and it gets converted before further validation) otherwise this just adds a lot more responsibilities for the validations that are string column specific as they will error out with numeric columns. 

Otherwise, will just have to add another if statement logic to check that the column is string or convert it to str because we won't be able to do our regex matching at all